src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,329 @@
+    """Recursively trotterize a QubitOperator according
+        to the scheme: e^(A+B) = e^(7/24 * A) e^(2/3 * B) e^(3/4 * A)
+        e^(-2/3 * B) e^(-1/24 * A) e^(B)
+
+    Note:
+        See N. Hatano and M. Suzuki Lect. Notes Phys 679, 37-68 (2005)
+
+    Args:
+        op_a: (QubitOperator) first term in hamiltonian
+        op_b: (list of QubitOperator) the rest of the terms in the
+                hamiltonian as a list of QubitOperators
+    Returns:
+        list of QubitOperators giving the trotterized hamiltonian
+    """
+    # Create list to hold return value
+    ret_val = []
+
+    # e^(7/24 * A)
+    ret_val.append(7.0/24.0 * op_a)
+
+    # e^(2/3 * B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(2.0 / 3.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(op * 2.0/3.0)
+        atemp = op_b[0] * 2.0/3.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    # e^(3/4 * A)
+    ret_val.append(3.0/4.0 * op_a)
+
+    # e^(-2/3 * B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(-2.0/3.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(op * -2.0/3.0)
+        atemp = op_b[0] * -2.0/3.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    # e^(-1/24 * A)
+    ret_val.append(-1.0/24.0 * op_a)
+
+    # e^(B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(1.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(1.0 * op)
+        atemp = op_b[0] * 1.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
+
+    return ret_val
+
+
+def get_trotterized_qubops(
+        hamiltonian,
+        trotter_number=None,
+        trotter_order=None,
+        term_ordering=None,
+        k_exp=None):
+    """Perform Trotter decomposition without exponentiating, to be
+    exponentiated later.
+
+    Args:
+        hamiltonian(QubitOperator): full hamiltonian
+        trotter_number(int): optional number of trotter steps -
+            default is 1
+        trotter_order(int): optional order of trotterization as
+            an integer from 1-3 - default is 1
+        term_ordering (list of (tuples of tuples)): optional list
+            of QubitOperator terms dictionary keys that specifies
+            order of terms when trotterizing
+        k_exp (float, int, or double): optional exponential factor
+            to all terms when trotterizing
+
+    Note:
+        The default term_ordering is simply the ordered keys of
+        the QubitOperators.terms dict.
+
+    Returns:
+        A list of single-Pauli-string QubitOperators
+
+    Raises:
+        ValueError if order > 3 or order <= 0,
+        TypeError for incorrect types
+    """
+
+    # Check for default arguments and type errors
+    if trotter_order is None:
```

Comment:
For the integer/float inputs, it's okay to just default them in the parameter line